scripts/ci/pre_commit/pre_commit_vendor_k8s_json_schema.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import json
+
+import requests
+
+K8S_DEFINITIONS = (
+    "https://raw.githubusercontent.com/yannh/kubernetes-json-schema"
+    "/master/v1.22.0-standalone-strict/_definitions.json"
+)
+VALUES_SCHEMA_FILE = "chart/values.schema.json"
+
+
+with open(VALUES_SCHEMA_FILE) as f:
+    schema = json.load(f)
+
+refs = set()
+
+
+def find_refs(props: dict):
+    for value in props.values():
+        if "$ref" in value:
+            refs.add(value["$ref"])
+
+        if "items" in value:
+            if "$ref" in value["items"]:
+                refs.add(value["items"]["$ref"])
+
+        if "properties" in value:
+            find_refs(value["properties"])
```

Comment:
```suggestion
def find_refs(props: dict):
    for value in props.values():
        if "$ref" in value:
            yield value["$ref"]

        if "items" in value:
            if "$ref" in value["items"]:
                yield value["items"]["$ref"]

        if "properties" in value:
            yield from find_refs(value["properties"])
```

Reply:
The code is just as long, but a tad more readable because the `refs` variable is created with a certain value and is not modified from another function.